### PR TITLE
EDM-165: Fleetless devices are editable

### PR DESCRIPTION
**Pull request by celdrake**
Fleetless devices are now editable.

- Extracted the common components related to the DeviceSpec out of the CreateFleetWizard.
- Small refactor to DRY helper text elements.

### CODE REVIEW

File: libs/ui-components/src/components/form/FormSelect.tsx [hunk 1]
```diff
@@ -72,33 +63,22 @@ const FormSelect: React.FC<FormSelectProps> = ({ name, items, helperText, placeh
...
-        {itemKeys.length && (
+        {itemKeys.length ? (
           <SelectList className="fctl-form-select__menu">
             {itemKeys.map((key) => (
               <SelectOption key={key} value={key}>
                 {items[key]}
               </SelectOption>
             ))}
           </SelectList>
+        ) : (
```
Comment (celdrake) [inline libs/ui-components/src/components/form/FormSelect.tsx]: A "0" was appearing when the dropdown was empty.

File: libs/ui-components/src/components/Device/EditDeviceWizard/EditDeviceWizard.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+          configTemplates: getConfigTemplatesValues(device.spec),
+        }}
+        validationSchema={getValidationSchema(t)}
+        validateOnMount
+        onSubmit={async (values) => {
+          setSubmitError(undefined);
+          try {
+            const patches = getDevicePatches(device, values);
+            if (patches.length > 0) {
+              await patch<Device>(`devices/${deviceId}`, getDevicePatches(device, values));
```
Comment (rawagner) [inline libs/ui-components/src/components/Device/EditDeviceWizard/EditDeviceWizard.tsx]:
```suggestion
              await patch<Device>(`devices/${deviceId}`, patches);
```

**Review by rawagner — CHANGES_REQUESTED** [review]
When there is a validation error on the second wizard page (Device Template), we can still jump to review page through the wizard navigation.

File: libs/ui-components/src/components/form/FormSelect.tsx [hunk 3]
```diff
@@ -72,33 +63,22 @@ const FormSelect: React.FC<FormSelectProps> = ({ name, items, helperText, placeh
           setIsOpen(open);
         }}
       >
-        {itemKeys.length && (
+        {itemKeys.length ? (
```
Comment (rawagner) [inline libs/ui-components/src/components/form/FormSelect.tsx]: this would be better
```suggestion
        {!!itemKeys.length &&  (
```

Comment (celdrake) [reply libs/ui-components/src/components/form/FormSelect.tsx]: That also works, thanks!

File: libs/ui-components/src/components/Device/EditDeviceWizard/EditDeviceWizard.tsx (hunk 2 above)
Comment (celdrake) [reply libs/ui-components/src/components/Device/EditDeviceWizard/EditDeviceWizard.tsx]: good catch

File: libs/ui-components/src/components/form/FormSelect.tsx (hunk 1 above)
Comment (celdrake) [reply libs/ui-components/src/components/form/FormSelect.tsx]: Fixed with Rasto's suggestion.

**Comment by celdrake** [comment]
> When there is a validation error on the second wizard page (Device Template), we can still jump to review page through the wizard navigation.

Should be fixed now.

**Review by rawagner — APPROVED** [review]

**Pull request merged**